src/components/Unknown/BuildScreen/Builder/CPUBuilder/index.tsx

```diff
@@ -52,24 +20,35 @@ const CPUBuilder: React.FC = () => {
     setExpand((prev) => !prev);
   };
 
-  const products = parts.map((part) => (
-    <BuilderProduct
-      product={part}
-      key={part.id}
-      handleSelect={handleAddProduct}
-      selectedId={selectedId}
-    />
-  ));
+  const firestore = useFirestore();
+
+  const productRef = firestore.collection('CPUs');
+
+  const { data, status } = useFirestoreCollectionData<CPU>(productRef);
+
+  const selectedProduct =
+    data && data.find((product) => product.id === selectedId);
 
   return (
     <ProductAccordion
       icon={CPUIcon}
       category={ProductCategoryByCollection.CPUs}
       selectedId={selectedId}
+      selectedProduct={
+        selectedProduct && normalizeProducts([selectedProduct])[0]
+      }
       expand={expand}
       toggleAccordion={toggleAccordion}
     >
-      {products}
+      {status === 'success' &&
+        normalizeProducts(data).map((product) => (
```

Comment:
normalizeProducts called two times, it will be better to save result of this func into variable
and selectedProduct will be variable too